### PR TITLE
Nfdiv 385/case created

### PR DESCRIPTION
**Pull request by velimd**
### JIRA link ###
https://tools.hmcts.net/jira/browse/NFDIV-452

### Change description ###
Call to create case when a user logs in.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/modules/api/cos-api.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { AxiosInstance } from 'axios';
+import { LoggerInstance } from 'winston';
+
+export class CosApi {
```
Comment (SebFlippence) [inline src/main/modules/api/cos-api.ts:4]: Perhaps we should start adding unit tests for new modules?

Comment (linusnorton) [reply src/main/modules/api/cos-api.ts:4]: this shouldn't be a module and it shouldn't be cos-api.ts :)

Comment (linusnorton) [reply src/main/modules/api/cos-api.ts:4]: modules should just be for express middleware setup

File: src/main/app/api/CosApi.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  //TODO change return type when backend has been implemented.
+  public getCase(): Promise<Record<string, unknown>> {
+    return this.axios
+      .get('/retrieve-case')
+      .then(results => results.data)
+      .catch(err => {
+        this.logger.error(err);
+        return false;
+      });
+  }
```
Comment (SebFlippence) [inline src/main/app/api/CosApi.ts:16]: Looks like this method needs a test to get it up [above the 60% coverage from sonarcloud](https://sonarcloud.io/component_measures?id=nfdiv-frontend&pullRequest=174&metric=new_coverage&view=list)

File: config/default.yaml [hunk 3]
```diff
@@ -8,6 +8,8 @@ services:
     tokenURL: 'https://idam-api.aat.platform.hmcts.net/o/token'
     clientID: 'divorce'
     clientSecret: 'NEED TO INSERT SECRET'
+  cos:
+    baseURL: 'http://localhost:4012'
```
Comment (SebFlippence) [inline config/default.yaml:12]: Do we need to add some instructions to the readme about how to get this service running locally?

Comment (velimd) [reply config/default.yaml:12]: included this however, you just need to run cos, nothing special config needed, let me know if i need to include more.

**Review by SebFlippence — APPROVED** [review]

File: charts/nfdiv-frontend/values.preview.template.yaml [hunk 4]
```diff
@@ -8,6 +7,9 @@ nodejs:
     prefix: civil
     primaryIngressHost: ${SERVICE_FQDN}
   environment:
+    IDAM_WEB_URL: 'https://idam-web-public.aat.platform.hmcts.net/login'
```
Comment (linusnorton) [inline charts/nfdiv-frontend/values.preview.template.yaml:10]: you don't need these three lines

**Review by linusnorton — APPROVED** [review]

**Review by SebFlippence — APPROVED** [review]

**Pull request merged**